### ISSUE TITLE
CLE Lab: Specimen Assessment pt.2

### ISSUE THREAD

**Issue by billtell** [GTAC-MGI/GTAC-ESP-LIMS#423]
This is a branch for the continuation of work begun on #307

**Comment by billtell** [comment]
- [x]  Confirm the format of the file that can / will be exported from nanodrop
> Excel / Adding csv format as well just in case
- [x] Error handling
> Adding `Exception` field to all CLE entities and having pipeline return "Fail Reason" back to `Exception` column

**Comment by billtell** [comment]
Changing `Sample Name` to `Accesssion Number`
Changing `Tissue Type` to `Sample Type`
This work was completed on #493

**Comment by billtell** [comment]
`Calculated Yield (ng)` changed to match entity value `Yield (ng)` on #493

**Comment by billtell** [comment]
Need to add protocol actions to sync `Assessed Concentration (ng/µl)` and `260/280` & `260/230`

**Comment by abohanno** [comment]
Hi Guys, I figured out how to annotate the Qubit Flex file today. We should now be able to scan the accession sticker to add it to the file prior to ESP upload. This can then act as a double check that the samples were in the right order. I will have the next person who does assessments label their files this way and I can export them and give them to you guys.

**Comment by billtell** [comment]
### bringing in pass/fail thresholds from #307 

**Concentration/Purity:**

Updated QC metrics for specimen assessment listed below:

IDT Tumor-Normal = Yield >250ng, 260/280 = 1.4 - 2.4
SOMA = Yield >250ng, 260/280 = 1.4 - 2.4
IDT Germline Exome = Yield >250ng, 260/280 = 1.4 - 2.4
Gateway Seq = Yield >250ng, 260/280 = 1.4 - 2.4
MyeloSeq = Yield >250ng, 260/280 = 1.4 - 2.4 (any MyeloSeq that is also a ChromoSeq should have >500ng)
ChromoSeq = Yield >250ng, 260/280 = 1.4 - 2.4 (any ChromoSeq that is also a MyeloSeq should have >500ng)
Positive Controls = Yield >250ng, 260/280 = 1.4 - 2.4
Negative Controls = Concentration <0.2ng/uL
Anything that does not meet the above conditions will need to be marked as a QC fail.

Things to consider for a possible error handler

Our preferred input for library construction is 500ng. Mark any samples between 250ng and 500ng.
Any blood or bone marrow specimen with less than 750ng should get another extraction in case we need more DNA. (Any blood or marrow that is both MyeloSeq and ChromoSeq will get another extraction if it has less than 1000ng)
May need to come up with an "aliquot status" column during accessioning or assessment. If status = true, the yield must also be greater than the amount required for NGS and aliquot.

**Volume:**

QIACube DNA (blood and bone marrow eluted in 100uL, 1 reaction): QC Range = 50uL - 125uL
QIACube DNA (blood and bone marrow eluted in 100uL, 2 reactions): QC Range = 150uL - 225uL
QIACube DNA (blood and bone marrow eluted in 100uL, 3 reactions): QC Range = 250uL - 325uL
QIACube DNA (blood and bone marrow eluted in 100uL, 4 reactions): QC Range = 350uL - 425uL
Manual Prep DNA (fresh tissue and FFPE eluted in 60uL): QC Range = 40uL – 70uL
Manual Prep DNA (Buccal Swabs eluted in 150uL, 2 reactions pooled): QC Range = 250uL – 310uL
Manual Prep DNA (cell pellets and fibroblasts eluted in 200uL): QC Range = 150uL – 225uL

**Comment by billtell** [comment]
- [x] `Assay `and `Sample Type` -> read_only
- [x] Remove Assessment Report tab
- remove from protocol
- remove from pipeline script
- [x] 2D Barcode check?

**Comment by billtell** [comment]
- [x] Hide instrument data values (volume, concentration) in extraction workflow sample protcol

```
function onrender_hide_cle_instrument_fields(api) {
	api.hideColumns(['Volume (µl)', 'Concentration (ng/µl)', 'Yield (ng)', '260/280', '260/230']); 
}
```

- added to `'CLE Lab: DNA Extraction'`

**Comment by billtell** [comment]
- Log Exception -> Exception on Specimen Type
- What is a Specimen Type Exception?
- Replacing `Note` entity value with `Exception`
- Note -> protocol property / Exception -> entity property

**Comment by billtell** [comment]
- [x] Make qubit file indexed on ESP ID , not order

**Comment by billtell** [comment]
- [x]  Add Volume QC
- [x] Set min/max for Yield and Purity.  Volume is too conditional to set a fixed min/max.

**Comment by billtell** [comment]
- fibroblasts sample type?

**Comment by billtell** [comment]
- [x] Simplify yield threshold lookup. 
>  If UDN Extractions -> 15000 , else -> 250 ?

**Comment by billtell** [comment]
### Dec. 3rd Questions:

1. How to handle manual extractions -> split off into separate workflow node
2. FFPE & Buccal Swabs = 2 tubes -> 1? -> create extraction pooling workflow to go to before assessment
3. Fibroblasts -> fibroblasts are cell pellets = disregard
4. Fail vs. Needs Review -> Needs Review isn't really needed. If value falls outside acceptable range, set to `Fail`

**Comment by billtell** [comment]
- [x] cDNA Exome be removed from list of assays
- [x] Add FNA Smear to Specimen Types

**Comment by billtell** [comment]
@abohanno I can't remember.  Did you say whether UDN samples get Specimen Assessment.  Yield is 15k ng, right?

**Comment by billtell** [comment]
@abohanno Can you share the Specimen Assessment Table or the link to your documentation so that I can make sure it is all captured in this pipeline script.  Like the one you shared for Library Assessment.

**Comment by billtell** [comment]
### Merge into `release-18`

```
Updating b3a66039..3c58965a
Fast-forward
 content/chains/cle_wf_chain.yml                                                                 | 140 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++-----------------------------
 content/inventory/entity_types.yml                                                              |  64 +++++++++++++++++++++++------------------
 content/pipelines/scripts/cle_specimen_assessment.py                                            | 152 +++++++++++++++++++++++++++++++++++++++----------------------------------------------------------
 content/protocols/{cle_make_extraction_protocol.yml => cle_auto_extraction_protocol.yml}        |   3 +-
 content/protocols/{cle_extraction_setup_protocol.yml => cle_auto_extraction_setup_protocol.yml} |   4 +--
 content/protocols/cle_manual_extraction_protocol.yml                                            |  35 +++++++++++++++++++++++
 content/protocols/cle_manual_extraction_setup_protocol.yml                                      |  76 +++++++++++++++++++++++++++++++++++++++++++++++++
 content/protocols/cle_specimen_assessment_protocol.yml                                          |   8 ++++++
 content/protocols/cle_specimen_assessment_review_protocol.yml                                   |  92 +++++++++++++++++++++++++++++++++++++++++++----------------
 content/workflows/{cle_dna_extraction_workflow.yml => cle_auto_extraction_workflow.yml}         |  18 ++++++------
 content/workflows/cle_manual_extraction_workflow.yml                                            |  21 ++++++++++++++
 extensions/client/invokables.js                                                                 |   4 +++
 roles/seed/content.yaml                                                                         |  18 ++++++------
 13 files changed, 428 insertions(+), 207 deletions(-)
 rename content/protocols/{cle_make_extraction_protocol.yml => cle_auto_extraction_protocol.yml} (82%)
 rename content/protocols/{cle_extraction_setup_protocol.yml => cle_auto_extraction_setup_protocol.yml} (98%)
 create mode 100644 content/protocols/cle_manual_extraction_protocol.yml
 create mode 100644 content/protocols/cle_manual_extraction_setup_protocol.yml
 rename content/workflows/{cle_dna_extraction_workflow.yml => cle_auto_extraction_workflow.yml} (51%)
 create mode 100644 content/workflows/cle_manual_extraction_workflow.yml
```

**Comment by abohanno** [comment]
Yes, UDN samples do get assessed. We should have >15ug for send out.
<img width="1629" alt="Screenshot 2024-12-02 at 1 12 57 PM" src="https://github.com/user-attachments/assets/3ab5f213-3d0b-4168-828e-2a62f8af4f0f">
